src/reducers/remove.js

```diff
@@ -8,9 +8,54 @@ import { Map, List } from 'immutable'
  * @return {Immutable.Map} - updated state
  */
 export const start = (state, action) => {
+  return state.withMutations(map => {
+    // Pessimistic removal
+    if (action.isAsync) {
+      // Add uid to pending.remove
+      const pendingRemove = map.getIn(['pending', 'remove'], List())
+      map.setIn(['pending', 'remove'], pendingRemove.insert(0, action.uid))
+    }
+    // Optimistic removal
```

Comment:
`// Optimistic removal` comment could be updated. We're really just removing the item here. Since no `remover` is added, this is the only and final step for removing an item.